artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTSessionState.java

```diff
@@ -186,35 +186,39 @@ public void setAttached(boolean attached) {
 
    public Collection<MqttTopicSubscription> getSubscriptions() {
       Collection<MqttTopicSubscription> result = new HashSet<>();
-      for (Pair<MqttTopicSubscription, Integer> pair : subscriptions.values()) {
-         result.add(pair.getA());
+      for (SubscriptionItem item : subscriptions.values()) {
+         result.add(item.getSubscription());
       }
       return result;
    }
 
    public Collection<Pair<MqttTopicSubscription, Integer>> getSubscriptionsPlusID() {
```

Comment:
I think this should just return `Map<String, MQTTSessionState.SubscriptionItem>`, e.g.:
```java
public Map<String, SubscriptionItem> getSubscriptionsPlusID() {
   return new HashMap<>(subscriptions);
}
```
It should be simply to update any code which calls this function to deal with the new return value.